datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -737,19 +743,33 @@ private void uninitializeTasks(List<DatastreamTask> tasks) {
     tasks.forEach(_cpProvider::unassignDatastreamTask);
   }
 
-  private void initializeTask(DatastreamTask task) {
-    DatastreamTaskImpl taskImpl = (DatastreamTaskImpl) task;
-    assignSerdes(taskImpl);
+  private void initializeTask(DatastreamTask task, boolean isDatastreamUpdate, boolean retryAndSaveError) {
+    try {
+      DatastreamTaskImpl taskImpl = (DatastreamTaskImpl) task;
+      assignSerdes(taskImpl);
 
-    boolean customCheckpointing = getCustomCheckpointing(task);
-    TransportProviderAdmin tpAdmin = _transportProviderAdmins.get(task.getTransportProviderName());
-    TransportProvider transportProvider = tpAdmin.assignTransportProvider(task);
-    EventProducer producer =
-        new EventProducer(task, transportProvider, _cpProvider, _eventProducerConfig, customCheckpointing);
+      boolean customCheckpointing = getCustomCheckpointing(task);
+      TransportProviderAdmin tpAdmin = _transportProviderAdmins.get(task.getTransportProviderName());
+      TransportProvider transportProvider = tpAdmin.assignTransportProvider(task);
+      EventProducer producer = new EventProducer(task, transportProvider, _cpProvider, _eventProducerConfig, customCheckpointing);
 
-    taskImpl.setEventProducer(producer);
-    Map<Integer, String> checkpoints = producer.loadCheckpoints(task);
-    taskImpl.setCheckpoints(checkpoints);
+      taskImpl.setEventProducer(producer);
+      Map<Integer, String> checkpoints = producer.loadCheckpoints(task);
+      taskImpl.setCheckpoints(checkpoints);
+    } catch (Exception e) {
```

Comment:
In the cases reported so far the specific exception type is "KafkaException" but I have a generic catch all exception here because all the reported cases were for LiKafkaTransportProvider and I wasn't sure if all transport providers would throw the same error. So to make it generic I have a catch all exception here, pls suggest the appropriate exception type is this is not correct.

Reply:
This is a good start, but evaluate the `InterruptedException` path as well. I think if the thread gets interrupted then we want to stop all the processing asap. 

Plz check CoordinatorEventProcessor code and the `handleEvent` call.

Reply:
I modified this to throw back the error up the call stack in case of InterruptedException so that it can fall back to the existing behavior in case of InterruptedException.

Reply:
Looks like the circleci failure for this patch is because of the InterruptedException at line 778.
The circleci warning is as follows 
"""
Bug type BC_IMPOSSIBLE_INSTANCEOF (click for details)In class com.linkedin.datastream.server.CoordinatorIn method com.linkedin.datastream.server.Coordinator.initializeTask(DatastreamTask, Set, boolean)Actual type RuntimeExceptionExpected InterruptedExceptionValue loaded from eAt Coordinator.java:[line 778]
"""
In the code if I change the code to add a explicit catch block for InterruptedException Intellij tells me that its not possible to receive InterruptedException in this block of code. I read up on it and looks like InterruptedException is not thrown immediately and its upto the thread to interrupt itself whenever convenient next and that is typically whenever a blocking call (like Thread.sleep) is made. So basically blocking calls can throw InterruptedException and in this block there is no blocking call so we dont need to worry about InterruptedException here. Plz correct me if I misunderstood it, if not I will update the patch and remove InterruptedException handling from here.

Reply:
I checked the `CoordinatorEventProcessor` code and looks like you don't need to worry about the `InterruptedException` since it checks whether the thread is interrupted before picking the message from the queue. This means even if a new message is added, it will be skipped while processing if the exception is InterruptedException.

Reply:
Ack. Will remove the InterruptedException handling from here.